packages/openid4vc-client/src/OpenId4VcClientService.ts

```diff
@@ -134,6 +160,42 @@ export class OpenId4VcClientService {
     }
   }
 
+  public generateCodeVerifier(): string {
+    return randomStringForEntropy(256)
+  }
+
+  public async generateAuthorizationUrl(options: GenerateAuthorizationUrlOptions) {
+    this.logger.debug('Generating authorization url')
+
+    if (!options.scope || options.scope.length === 0) {
+      throw new AriesFrameworkError(
+        'Only scoped based authorization requests are supported at this time. Please provide at least one scope'
+      )
+    }
+
+    const client = await OpenID4VCIClient.initiateFromURI({
+      issuanceInitiationURI: options.initiationUri,
+      flowType: AuthzFlowType.AUTHORIZATION_CODE_FLOW,
+    })
+
+    const hashed = sha256(TypedArrayEncoder.fromString(options.codeVerifier))
+    const base64Url = TypedArrayEncoder.toBase64URL(Buffer.from(hashed))
+
+    this.logger.debug('Converted code_verifier to code_challenge', {
+      codeVerifier: options.codeVerifier.toString(),
+      sha256: hashed.toString(),
```

Comment:
What encoding must the hash have? Utf8?

Reply:
The hash is base64 URL encoded (see `codeVerifierSha256`) variable. Renamed some variables for clarity.

Reply:
Does to string automatically encode it as base64?

Reply:
I'm always a bit worried with the toString method as it's a method available on all objects and will ofyen give you `[Object object]` or something